pandas/core/generic.py

```diff
@@ -1395,7 +1396,7 @@ def _set_axis_name(self, name, axis=0, inplace=False):
     # ----------------------------------------------------------------------
     # Comparison Methods
 
-    def _indexed_same(self, other):
+    def _indexed_same(self, other) -> bool:
```

Comment:
why do we have bool here and above but bool_t below?  i expected to need bool_t throughout

also i think `other` should have same type as self here

Reply:
Hmm yea curious what the pattern here is; do only things after `def bool` need the alias? Or do you know if this is no longer an issue with Py36 onwards?

Reply:
That's what appears to be the case. `def bool` is in the class namespace and doesn't effect the use of bool inside other methods see L8658. However, the functions signatures must be evaluated in the class namespace since any types added to the function signatures after the `def bool` need to use the alias.

Reply:
> also i think `other` should have same type as self here

hmm, the types should be representative of the runtime behaviour. should `pd.Series()._indexed_same(pd.DataFrame())` raise?

```
>>> pd.__version__
'0.26.0.dev0+1155.ged20822a5'
>>>
>>> pd.DataFrame()._indexed_same(pd.DataFrame())
True
>>>
>>> pd.Series()._indexed_same(pd.DataFrame())
True
>>>
>>> pd.Series()._indexed_same(pd.Series())
True
>>>
>>> pd.DataFrame()._indexed_same(pd.Series())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\simon\pandas\pandas\core\generic.py", line 1400, in _indexed_same
    self._get_axis(a).equals(other._get_axis(a)) for a in self._AXIS_ORDERS
  File "C:\Users\simon\pandas\pandas\core\generic.py", line 1400, in <genexpr>
    self._get_axis(a).equals(other._get_axis(a)) for a in self._AXIS_ORDERS
  File "C:\Users\simon\pandas\pandas\core\generic.py", line 444, in _get_axis
    name = self._get_axis_name(axis)
  File "C:\Users\simon\pandas\pandas\core\generic.py", line 441, in _get_axis_name
    raise ValueError("No axis named {0} for object type {1}".format(axis, cls))
ValueError: No axis named columns for object type <class 'pandas.core.series.Series'>
>>>
```

Reply:
I'd ask for bool_t to be made consistent across the module, totally OK with leaving `other` here as out of scope